lib/puppet/module.rb

```diff
@@ -128,6 +128,8 @@ def has_metadata?
       end
 
       return nil unless Puppet::FileSystem.exist?(full_path)
+      # TODO: line left in while testing to see if this is necessary
+      Puppet.warning("Unexpanded path #{full_path}") if full_path.upcase != Puppet::FileSystem.expanded_path_string(full_path).upcase
```

Comment:
This code path is not used (IIRC) from the 4.x loaders, and as comments said this is used primarily for file serving, so I don't have anything to say here. Hope that piece of code is not needed, and if needed users should have done the "expand" piece earlier if that is required.